src/wallets/injected/SenderWallet.ts

```diff
@@ -92,14 +88,9 @@ class SenderWallet implements InjectedWallet {
       await this.init();
     }
 
-    const rpcResponse = await this.wallet.getRpc();
-
-    if (!this.networkMatches(rpcResponse)) {
-      return;
-    }
-
     const { accessKey } = await this.wallet.requestSignIn({
       contractId: this.options.contract.accountId,
+      methodNames: this.options.contract.changeMethods || [],
```

Comment:
Is `methodNames` needed? 

It impacts whether the same function call key can be reused or not, once some new methodsNames are added or removed.

Reply:
I was experimenting with the options API for contracts but hadn't implemented `changeMethods` or `viewMethods`.

I did some digging and found both Sender Wallet and NEAR Wallet use `methodNames` instead of having two parameters. I'd omit this change for now and we can raise another PR in the future that makes use of this option for all the wallets to avoid inconsistency.

Reply:
yes. `methodNames` are used in `requestSignIn()`, and `changeMethods` / `viewMethods` are used when initializing the contract instance.